diff --git a/blarser/src/schema.rs b/blarser/src/schema.rs
index 3a80c7b..750fd25 100644
--- a/blarser/src/schema.rs
+++ b/blarser/src/schema.rs
@@ -81,17 +81,39 @@ table! {
         event_aux_data -> Jsonb,
         observations -> Array<Timestamptz>,
         terminated -> Nullable<Text>,
     }
 }
 
+table! {
+    use diesel::sql_types::*;
+    use crate::db_types::*;
+    use crate::state::Event_source;
+
+    versions_with_end (id) {
+        id -> Int4,
+        ingest_id -> Int4,
+        entity_type -> Text,
+        entity_id -> Uuid,
+        start_time -> Timestamptz,
+        end_time -> Nullable<Timestamptz>,
+        entity -> Jsonb,
+        from_event -> Int4,
+        event_aux_data -> Jsonb,
+        observations -> Array<Timestamptz>,
+        terminated -> Nullable<Text>,
+    }
+}
+
 joinable!(event_effects -> events (event_id));
 joinable!(versions -> events (from_event));
+joinable!(versions_with_end -> events (from_event));
 
 allow_tables_to_appear_in_same_query!(
     approvals,
     event_effects,
     events,
     ingests,
     version_links,
     versions,
+    versions_with_end,
 );
